### PR TITLE
Prepare version 0.15.0

### PR DESCRIPTION
**Pull request by JayH5**
- [x] Write the changelog
- [x] #1157 
- [x] #1201 
- [x] Anything else we want to squeeze in?? #1200 ✅ 
- [x] Bump the version

### CODE REVIEW

**Comment by graingert** [comment]
You'll probably want to bump the anyio dep to 3.2 to get py3.10 support

**Comment by JayH5** [comment]
> You'll probably want to bump the anyio dep to 3.2 to get py3.10 support

ack, but I'm not sure any of anyio 3.2's Python 3.10-compatibility fixes apply directly to Starlette--our test suite passes. I've written "Initial support" in the changelog and wouldn't want to hold up the Starlette release for an AnyIO release.

**Comment by graingert** [comment]
Ok that seems reasonable

**Comment by JayH5** [comment]
AnyIO 3.2.0 is out now.

**Review by graingert — APPROVED** [review]

File: docs/release-notes.md
Comment (Kludex) [inline docs/release-notes.md]: It's a draft, but as it was already reviewed, I'll just add a reminder so you don't forget to remove this line.

**Comment by JayH5** [comment]
@encode/maintainers since I've made a few changes here since approval, can I get one last ✅, please?

**Review by graingert — APPROVED** [review]

**Review by Kludex — APPROVED** [review]

**Pull request merged**
